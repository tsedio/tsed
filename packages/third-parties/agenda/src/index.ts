/**
 * @file Automatically generated by barrelsby.
 */

export * from "./AgendaModule.js";
export * from "./constants/constants.js";
export * from "./decorators/agenda.js";
export * from "./decorators/define.js";
export * from "./decorators/every.js";
export * from "./interfaces/AgendaStore.js";
export * from "./interfaces/interfaces.js";
export * from "./services/AgendaFactory.js";
