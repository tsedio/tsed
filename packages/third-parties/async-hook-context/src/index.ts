export * from "./decorators/injectContext";
export * from "./services/PlatformAsyncHookContext";
