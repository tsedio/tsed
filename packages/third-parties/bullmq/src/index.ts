/**
 * @file Automatically generated by barrelsby.
 */

export * from "./BullMQModule";
export * from "./config/config";
export * from "./constants/BullMQTypes";
export * from "./constants/constants";
export * from "./contracts/JobMethods";
export * from "./contracts/JobStore";
export * from "./decorators/InjectQueue";
export * from "./decorators/InjectWorker";
export * from "./decorators/JobController";
export * from "./dispatchers/JobDispatcher";
export * from "./dispatchers/JobDispatcherOptions";
export * from "./utils/createQueueProvider";
export * from "./utils/createWorkerProvider";
export * from "./utils/getJobToken";
export * from "./utils/getQueueToken";
export * from "./utils/getWorkerToken";
export * from "./utils/mapQueueOptions";
export * from "./utils/mapWorkerOptions";
