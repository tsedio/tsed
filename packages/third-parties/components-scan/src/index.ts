/**
 * @file Automatically generated by barrelsby.
 */

export * from "./cleanGlobPatterns.js";
export * from "./importFiles.js";
export * from "./importProviders.js";
export * from "./isTsEnv.js";
