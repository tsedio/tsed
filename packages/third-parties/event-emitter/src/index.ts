import {ConstructorOptions} from "eventemitter2";

declare global {
  namespace TsED {
    interface Configuration {
      eventEmitter?: {
        /**
         * Enable Event Emitter. Default `false`.
         */
        enabled?: boolean;
        /**
         * Disable event table displayed in the logger. Default `false`.
         */
        disableSummary?: boolean;
      } & ConstructorOptions;
    }
  }
}

export * from "./EventEmitterModule";
export * from "./services/EventEmitterFactory";
export * from "./decorators";
