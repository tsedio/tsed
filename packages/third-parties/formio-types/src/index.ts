/**
 * @file Automatically generated by barrelsby.
 */

export * from "./domain/FormioAccess.js";
export * from "./domain/FormioAction.js";
export * from "./domain/FormioActionInfo.js";
export * from "./domain/FormioActionItem.js";
export * from "./domain/FormioComponent.js";
export * from "./domain/FormioForm.js";
export * from "./domain/FormioPermission.js";
export * from "./domain/FormioRole.js";
export * from "./domain/FormioSchema.js";
export * from "./domain/FormioSubmission.js";
export * from "./domain/FormioToken.js";
