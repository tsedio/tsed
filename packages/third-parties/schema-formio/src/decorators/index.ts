export * from "./component";
export * from "./currency";
export * from "./dataSourceJson";
export * from "./dataSourceUrl";
export * from "./hidden";
export * from "./inputTags";
export * from "./multiple";
export * from "./password";
export * from "./select";
export * from "./tableView";
export * from "./textarea";
export * from "./errorMessage";
export * from "./errorLabel";
export * from "./minWords";
export * from "./maxWords";
export * from "./validate";
export * from "./form";
export * from "./tabs";
export * from "./placeholder";
export * from "./tooltip";
export * from "./label";
export * from "./customClass";
export * from "./modalEdit";
export * from "./mask";
export * from "./prefix";
export * from "./suffix";
export * from "./textCase";
export * from "./conditional";
export * from "./customConditional";
