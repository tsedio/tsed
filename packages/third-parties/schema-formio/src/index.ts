export * from "./components";
export * from "./decorators";
export * from "./registries/FormioMappersContainer";
export * from "./utils/getFormioSchema";
