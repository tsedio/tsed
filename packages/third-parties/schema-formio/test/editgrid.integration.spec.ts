import {CollectionOf, Property} from "@tsed/schema";
import {Currency, getFormioSchema} from "../src";


describe('EditGrid integration', () => {
  it("should generate form", () => {
    class Nested {
      @Property()
      id: string;
    }
    class Model {
      @Currency()
      @CollectionOf(Nested)
      test: Nested[];
    }

    expect(getFormioSchema(Model)).toEqual({
      components: [
        {
          components: [
            {
              disabled: false,
              input: true,
              key: "id",
              label: "Id",
              type: "textfield",
              validate: {
                required: false
              }
            }
          ],
          disabled: false,
          input: true,
          key: "test",
          label: "Test",
          rowDrafts: false,
          type: "editgrid",
          validate: {
            required: false
          }
        }
      ],
      display: "form",
      machineName: "model",
      name: "model",
      title: "Model",
      type: "form",
      submissionAccess: [],
      access: [],
      tags: [],
    });
  });
})
