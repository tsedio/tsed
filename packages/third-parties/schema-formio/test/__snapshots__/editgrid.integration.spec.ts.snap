// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditGrid integration should generate form with minItems/maxItems 1`] = `
Object {
  "access": Array [],
  "components": Array [
    Object {
      "components": Array [
        Object {
          "disabled": false,
          "input": true,
          "key": "id",
          "label": "Id",
          "type": "textfield",
          "validate": Object {
            "required": false,
          },
        },
      ],
      "disabled": false,
      "input": true,
      "key": "test",
      "label": "Test",
      "rowDrafts": false,
      "type": "editgrid",
      "validate": Object {
        "max": 10,
        "min": 1,
        "required": false,
      },
    },
  ],
  "display": "form",
  "machineName": "model",
  "name": "model",
  "submissionAccess": Array [],
  "tags": Array [],
  "title": "Model",
  "type": "form",
}
`;
